src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -872,11 +870,15 @@ public void Initialize()
             _engine.Initialize(_env, Configuration, clusterChanges, myUrl, Server.Time, out _lastClusterTopologyIndex, ServerShutdown);
 
             using (Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
-            using (context.OpenReadTransaction())
+            using (var wtx = context.OpenWriteTransaction())
             {
                 PublishedServerUrls = PublishedServerUrls.Read(context);
-
-                LoadDefaultIdentityPartsSeparator(context);
+                var clusterState = (ClusterStateRecord)wtx.InnerTransaction.LowLevelTransaction.CurrentStateRecord.ClientState;
+                wtx.InnerTransaction.LowLevelTransaction.UpdateClientState(clusterState with
+                {
+                    DefaultIdentityPartsSeparator = LoadDefaultIdentityPartsSeparator(context)
+                });
+                wtx.Commit();
```

Comment:
Because all our state is now held as a `ClientState` inside the transaction, not outside of it.
So we need to commit that.
In practice, this changes no pages, so just update the in memory stae.